foodji_api/DomainTests/Recipes/RecipeTests.cs

```diff
@@ -54,6 +80,58 @@ public void GivenValidValues_Create_ReturnsRecipe()
             actualRecipe.Author.Should().Be(expectedAuthor);
             actualRecipe.Should().BeOfType<Recipe>();
         }
+
+        [Test]
+        public void GivenValidRecipeToUpdate_UpdateRecipe_RecipeUpdated()
+        {
+            // Arrange new substitute
+            var substitute = RecipeSubstitute.Create(
+                "updatedSubstituteName",
+                "updatedSubstituteDescription",
+                "substituteDescription",
+                Measurement.Create(
+                    1,
+                    UnitType.Millilitre,
+                    "updated alternative text"),
+                new List<Tag> { Tag.Vegan });
+                
+            // Arrange new ingredient
+            var ingredient = RecipeIngredient.Create(
+                "updatedIngredientName",
+                "updatedIngredientDescription",
+                Measurement.Create(
+                    2,
+                    UnitType.Gram,
+                    "updated alternative text"),
+                new List<Tag> { Tag.Vegan },
+                new List<RecipeSubstitute> { substitute });
+
+            // Arrange new recipe
+            var expectedRecipe = Recipe.Create(
+                "expectedName",
+                RecipeCategory.Dessert,
+                "expectedDescription",
+                RecipeDetails.Create(1,1,1,1),
+                new List<RecipeIngredient> { ingredient },
+                new List<string> { "updatedRecipeStep" },
+                new Uri("https://www.yahoo.ca"),
+                "updatedAuthor");
+            
+            // Act
+            _recipe.Update(expectedRecipe);
+            
+            // Assert
+            // I know we should be testing the Id here,
+            // but I'm not sure how to do that since Create() does not instantiate the Id property...
```

Comment:
Je ne penses pas qu'on puisse tester l'Id ici sans faire un set manuel, et pour ça, faudrait soit donner un setter public à l'id (I'd rather not), ou y aller par réflexion, which *could* be worthwhile learning, mais à voir si c'est de quoi qu'on veut faire.

Reply:
On le donne pas, certes, mais faut pas oublier que ces tests sont pour vérifier les changements que *nous* on fait au code, donc ça reste valide - c'est une question de pas se tromper pis passer vite pis ajouter un update au ID parce qu'on est un peu dans la lune sur un prochain refactor.

Pour ce qui est de faire un refactor pour permettre de faire le test, comme je mentionnes, je suis pas down non plus, hence le pourquoi si on veux vérifier, je passerais par réflection, ce qui permet de bypass l'aspect private, mais je suis pas exactement calée sur le sujet en termes d'implémentation.

Bref, on *peut* skip, mais faut être au courant que ça reste techniquement une faille.